src/Forms/GridField/GridFieldEditButton.php

```diff
@@ -27,9 +27,37 @@ class GridFieldEditButton implements GridField_ColumnProvider
     protected $extraClass = [
         'grid-field__icon-action--hidden-on-hover' => true,
         'font-icon-edit' => true,
-        'btn--icon-large' => true
+        'btn--icon-large' => true,
+        'action-menu--handled' => true
     ];
 
+    public function getTitle($gridField, $record)
+    {
+        return 'Edit';
+    }
+
+    public function getGroup($gridField, $record)
+    {
+        return 'Default';
+    }
+
+    public function getType($gridField, $record)
+    {
+        return GridField_ActionMenuItem::LINK;
+    }
+
+    public function getUrl($gridField, $record)
+    {
+        return Controller::join_links($gridField->Link('item'), $record->ID, 'edit');
+    }
+
+    public function getExtraData($gridField, $record, $columnName)
+    {
+        return [
+            "classNames" => "action-detail edit-link"
+        ];
```

Comment:
Mix of short and long array syntax throughout this whole class. If you're adding new array definitions, might as well go through and fix all the others to keep it uniform.